platform/security/rest/seceurity-rest-authentication/src/main/java/org/codice/ddf/security/rest/authentication/AuthenticationEndpoint.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.rest.authentication;
+
+import java.net.URI;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+
+import org.apache.cxf.ws.security.tokenstore.SecurityToken;
+import org.codice.ddf.security.handler.api.BaseAuthenticationToken;
+import org.codice.ddf.security.handler.api.UPAuthenticationToken;
+import org.codice.ddf.security.policy.context.ContextPolicy;
+import org.codice.ddf.security.policy.context.ContextPolicyManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.common.SecurityTokenHolder;
+import ddf.security.http.SessionFactory;
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+
+@Path("/")
+public class AuthenticationEndpoint {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthenticationEndpoint.class);
+
+    @Context
+    UriInfo uriInfo;
+
+    private ContextPolicyManager contextPolicyManager;
+
+    private SecurityManager securityManager;
+
+    private SessionFactory sessionFactory;
+
+    public AuthenticationEndpoint(ContextPolicyManager policyManager,
+            SecurityManager securityManager, SessionFactory sessionFactory) {
+        this.contextPolicyManager = policyManager;
+        this.securityManager = securityManager;
+        this.sessionFactory = sessionFactory;
+    }
+
+    @POST
+    public Response login(@Context HttpServletRequest request,
+            @FormParam("username") String username, @FormParam("password") String password,
+            @FormParam("prevurl") String prevurl) throws SecurityServiceException {
+
+        // Make sure we're using HTTPS
+        if (!request.isSecure()) {
+            throw new IllegalArgumentException("Authentication request must use TSL.");
```

Comment:
Ah, we reject the request if it's not using TLS, good. Also, typo: `TSL` instead of `TLS`